gc/bte/bte.h

```diff
@@ -120,34 +142,47 @@ typedef s32 (*btecallback)(s32 result,void *userdata);
 
 void BTE_Init(void);
 void BTE_Shutdown(void);
+void BTE_Close(void);
 s32 BTE_InitCore(btecallback cb);
 s32 BTE_ApplyPatch(btecallback cb);
 s32 BTE_InitSub(btecallback cb);
 s32 BTE_ReadStoredLinkKey(struct linkkey_info *keys,u8 max_cnt,btecallback cb);
 s32 BTE_ReadBdAddr(struct bd_addr *bdaddr, btecallback cb);
+s32 BTE_SetEvtFilter(u8 filter_type,u8 filter_cond_type,u8 *cond, btecallback cb);
+s32 BTE_ReadRemoteName(struct pad_name_info *padinfo, btecallback cb);
+s32 BTE_Inquiry(u8 max_cnt,u8 flush, btecallback cb);
```

Comment:
We should not use the existing btecallback, because that forces us to pass the result in the `userdata` parameter, therefore making it impossible for the client to receive its own "userdata" pointer. I suggest using this callback type:

```c
typedef s32 (*bte_inquiry_callback)(s32 result, bte_inquiry_res *res, void *userdata);
```

or, actually better, move the s32 result inside bte_inquiry_res (or we can change `count` to be `s32`, and a negative value means error) and have

```c
typedef s32 (*bte_inquiry_callback)(bte_inquiry_res *res, void *userdata);
```

I haven't check the rest of the new functions, but please consider allowing the client to receive its own userdata.